crypto/kem/README.md

```diff
@@ -0,0 +1,229 @@
+# AWS-LC KEM design
+
+## Key Encapsulation Mechanism (KEM)
+
+The purpose of a KEM is to establish a shared secret between two parties. There are three main functionalities defined for a KEM: key generation, encapsulation, decapsulation. The protocol works in three steps:
+
+```
+1. Alice generates her key pair (public_key, secret_key). Sends public_key to Bob.
+2. Bob generates a shared secret and encapsulates it using Alices public key.
+   This produces a ciphertext. Sends ciphertext to Alice.
+3. Alice decapsulates the ciphertext with her key pair and obtains the shared secret.
+   Alices and Bobs shared secrets are the same if the protocol succeeded.
+```
+
+The KEM design in AWS-LC is generic and not specific to any particular KEM. The primary use-case for the API are the post-quantum cryptographic algorithms that were specified and evaluated in NIST's Post-Quantum Crypto Standardization Project ([link](https://csrc.nist.gov/Projects/post-quantum-cryptography/post-quantum-cryptography-standardization)).
+
+## The public KEM API
+
+**Core functions.** To use a KEM, you need to have access to three core functions: key generation, encapsulation, decapsulation. Following the general design in AWS-LC, these these three functions are offered in the form of standard AWS-LC EVP APIs:
+
+```
+// Generates a (public, private) key pair
+EVP_PKEY_keygen(ctx, &pkey);
+
+// KEM Encapsulation
+EVP_PKEY_encapsulate(ctx, ciphertext, &ciphertext_len,
+                          shared_secret, &shared_secret_len);
+
+// KEM Decapsulation
+EVP_PKEY_decapsulate(ctx, shared_secret, &shared_secret_len,
+                          ciphertext, ciphertext_len);
+```
+
+The `ctx` variable above is a pointer to a “context” object of type `EVP_PKEY_CTX`. The `pkey` variable listed as an argument of `EVP_PKEY_keygen` function is a pointer to an object of type `EVP_PKEY`. The `ciphertext` and `shared_secret` are byte arrays of length `ciphertext_len` and `shared_secret_len`, respectively.
+
+**The context.** You to create the context. Depending on the use case, this can be done in two ways:
```

Comment:
I'm not sure what you're trying to say here with "You to create", should it be "How to create"?

Reply:
yep, typo, fixed.